src/main/scala/FilterCaptcha.scala

```diff
@@ -0,0 +1,68 @@
+import com.sksamuel.scrimage._
+import com.sksamuel.scrimage.filter._
+import java.awt.image.BufferedImage
+import java.awt.{Graphics2D,Color,Font}
+
+class FilterCaptcha extends CaptchaProvider {
+  val tokenAnswer = scala.collection.mutable.Map[String, String]()
+  def getChallenge(): Challenge = {
+    val filterTypes = List(new FilterType1, new FilterType2)
+    val r = new scala.util.Random
+    val alphabet = "abcdefghijklmnopqrstuvwxyz"
+    val n = 8
+    val answer = Stream.continually(r.nextInt(alphabet.size)).map(alphabet).take(n).mkString
+    val token = scala.util.Random.nextInt(10000).toString
+    tokenAnswer += token -> answer
```

Comment:
You can encase just this line with `synchronized` to make this function thread safe.